logdna/resource_stream_exclusion.go

```diff
@@ -0,0 +1,178 @@
+package logdna
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type streamExclusion struct {
+	ID     string   `json:"id,omitempty"`
+	Title  string   `json:"title,omitempty"`
+	Active bool     `json:"active,omitempty"`
+	Apps   []string `json:"apps,omitempty"`
+	Hosts  []string `json:"hosts,omitempty"`
+	Query  string   `json:"query,omitempty"`
```

Comment:
I never liked the use of `omitempty`, but it stayed in the legacy code cause it would break the previous implementation.  Is the streams api response consistent enough to where the fields are always present?  We should remove this IMO if we can since we want the shape to remain consistent, no?

Reply:
good point, I think we could do that 👍

Reply:
I pulled out the `omitempty` everywhere except the properties that are prohibited from sending